lib/Sema/SemaDecl.cpp

```diff
@@ -8180,6 +8180,7 @@ void Sema::CheckMain(FunctionDecl* FD, const DeclSpec& DS) {
           Context.hasSameType(QualType(qs.strip(PT->getPointeeType()), 0),
                               Context.CharTy)) {
         qs.removeConst();
+        qs.removeAddressSpace();
```

Comment:
We don't want to allow arbitrary address spaces.  We should check that the AS matches the default AS (see ASTContext).